rclcpp_action/test/test_server.cpp

```diff
@@ -923,3 +982,249 @@ TEST_F(TestServer, deferred_execution)
   received_handle->execute();
   EXPECT_TRUE(received_handle->is_executing());
 }
+
+class TestValidServer : public TestServer
+{
+public:
+  TestValidServer()
+  : node_(nullptr), action_server_(nullptr), current_goal_handle_(nullptr) {}
+
+  void SetUp()
+  {
+    node_ = std::make_shared<rclcpp::Node>("is_ready_error", "/rclcpp_action/construct");
+    using GoalHandle = rclcpp_action::ServerGoalHandle<Fibonacci>;
+    action_server_ = rclcpp_action::create_server<Fibonacci>(
+      node_, "fibonacci",
+      [](const GoalUUID &, std::shared_ptr<const Fibonacci::Goal>) {
+        return rclcpp_action::GoalResponse::ACCEPT_AND_DEFER;
+      },
+      [](std::shared_ptr<GoalHandle>) {
+        return rclcpp_action::CancelResponse::REJECT;
+      },
+      [this](std::shared_ptr<GoalHandle> handle) {
+        current_goal_handle_ = handle;
+      });
+  }
+
+protected:
+  std::shared_ptr<rclcpp::Node> node_;
+  std::shared_ptr<rclcpp_action::Server<Fibonacci>> action_server_;
+  std::shared_ptr<rclcpp_action::ServerGoalHandle<Fibonacci>> current_goal_handle_;
+};
+
+TEST_F(TestValidServer, is_ready_rcl_error) {
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  auto rcl_context = node_->get_node_base_interface()->get_context()->get_rcl_context().get();
+  ASSERT_EQ(
+    RCL_RET_OK,
+    rcl_wait_set_init(&wait_set, 10, 10, 10, 10, 10, 10, rcl_context, allocator));
+  RCLCPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_wait_set_fini(&wait_set));
+  });
+  ASSERT_TRUE(action_server_->add_to_wait_set(&wait_set));
+
+  EXPECT_TRUE(action_server_->is_ready(&wait_set));
+  auto mock = mocking_utils::patch_and_return(
+    "lib:rclcpp_action", rcl_action_server_wait_set_get_entities_ready, RCL_RET_ERROR);
+  EXPECT_THROW(action_server_->is_ready(&wait_set), rclcpp::exceptions::RCLError);
+}
+
+enum class RequestType {GOAL, CANCEL};
+
+class TestBasicServer : public TestServer
+{
+protected:
+  void SetupActionServerAndSpin(
+    RequestType request_type,
+    std::chrono::milliseconds timeout = std::chrono::milliseconds(1000))
+  {
+    using GoalHandle = rclcpp_action::ServerGoalHandle<Fibonacci>;
+    if (nullptr != node_) {
+      node_.reset();
+    }
```

Comment:
@brawner nit: if `SetupActionServerAndSpin` returned all relevant state instead of persisting it, preventive resets would not be necessary.